superset/views/core.py

```diff
@@ -64,7 +64,14 @@
 ACCESS_REQUEST_MISSING_ERR = __(
     'The access requests seem to have been deleted')
 USER_MISSING_ERR = __('The user seems to have been deleted')
-DATASOURCE_ACCESS_ERR = __("You don't have access to this datasource")
+perms_instruction_link = config.get('PERMISSION_INSTRUCTIONS_LINK')
+if perms_instruction_link:
+    DATASOURCE_ACCESS_ERR = __(
+        "You don't have access to this datasource. <a href='{}'>How to gain access.</a>"
```

Comment:
I think this is fine. If you wanted to truncate the length of the message maybe something like 
> You don't have access to this datasource (Gain Access)